test/benchmarks/utils.py

```diff
@@ -1,168 +1,117 @@
-            index=index,
-        )
-        assert document_store.get_document_count() == 0
-    elif document_store_type in ("opensearch_flat", "opensearch_hnsw"):
-        launch_opensearch(local_port=9201)
-        if document_store_type == "opensearch_flat":
-            index_type = "flat"
-        elif document_store_type == "opensearch_hnsw":
-            index_type = "hnsw"
-        document_store = OpenSearchDocumentStore(index_type=index_type, port=9201, timeout=3000)
-    else:
-        raise Exception(f"No document store fixture for '{document_store_type}'")
-    return document_store
-
-
-def get_retriever(retriever_name, doc_store, devices):
-    if retriever_name == "elastic":
-        return BM25Retriever(doc_store)
-    if retriever_name == "tfidf":
-        return TfidfRetriever(doc_store)
-    if retriever_name == "dpr":
-        return DensePassageRetriever(
-            document_store=doc_store,
-            query_embedding_model="facebook/dpr-question_encoder-single-nq-base",
-            passage_embedding_model="facebook/dpr-ctx_encoder-single-nq-base",
-            use_gpu=True,
-            use_fast_tokenizers=False,
-            devices=devices,
-        )
-    if retriever_name == "sentence_transformers":
-        return EmbeddingRetriever(
-            document_store=doc_store,
-            embedding_model="nq-distilbert-base-v1",
-            use_gpu=True,
-            model_format="sentence_transformers",
+def prepare_environment(pipeline_config: Dict, benchmark_config: Dict):
+    """
+    Prepare the environment for running a benchmark.
+    """
+    # Download data if specified in benchmark config
+    if "data_url" in benchmark_config:
+        download_from_url(url=benchmark_config["data_url"], target_dir="data/")
+
+    n_docs = 0
+    if "documents_directory" in benchmark_config:
+        documents_dir = Path(benchmark_config["documents_directory"])
+        n_docs = len(
+            [
+                file_path
+                for file_path in documents_dir.iterdir()
+                if file_path.is_file() and not file_path.name.startswith(".")
+            ]
         )
 
+    # Launch DocumentStore Docker container if needed
+    for comp in pipeline_config["components"]:
+        if comp["type"].endswith("DocumentStore"):
+            launch_document_store(comp["type"], n_docs=n_docs)
+            break
+
 
-def get_reader(reader_name, reader_type, max_seq_len=384):
-    reader_class = None
-    if reader_type == "farm":
-        reader_class = FARMReader
-    elif reader_type == "transformers":
-        reader_class = TransformersReader
-    return reader_class(reader_name, top_k_per_candidate=4, max_seq_len=max_seq_len)
+def launch_document_store(document_store: str, n_docs: int = 0):
+    """
+    Launch a DocumentStore Docker container.
+    """
+    java_opts = None if n_docs < 500000 else "-Xms4096m -Xmx4096m"
+    if document_store == "ElasticsearchDocumentStore":
```

Comment:
If you want we can make these callables. In a benchmark setup part, we can register all document stores and their run callables and then here execute those callables.